### PR TITLE
Add x username to profile lookup

### PR DESCRIPTION
**Pull request by musidlo**

### CODE REVIEW

File: supabase/migrations/20241125111812_add-x-username-to-profile-lookup.sql [hunk 1]
```diff
@@ -0,0 +1,47 @@
+drop function if exists "public"."profile_lookup"(lookup_type lookup_type_enum, identifier text);
+
+set check_function_bodies = off;
+
+CREATE OR REPLACE FUNCTION public.profile_lookup(lookup_type lookup_type_enum, identifier text)
+ RETURNS TABLE(id uuid, avatar_url text, name text, about text, refcode text, x_username text, tag citext, address citext, chain_id integer, is_public boolean, sendid integer, all_tags text[])
+ LANGUAGE plpgsql
+ IMMUTABLE SECURITY DEFINER
```
[the remark this answers is not in the document]
Comment (musidlo) [reply supabase/migrations/20241125111812_add-x-username-to-profile-lookup.sql:8]: didnt change that, thats how this function was defined

**Review by youngkidwarrior — CHANGES_REQUESTED** [review]
This looks good on first glance but I'm gonna come back to it a second time when I'm not so busy. 

If @0xBigBoss has time for review then we can merge

This is a staple function in the codebase and I need more time to read it and learn it

**Comment by musidlo** [comment]
> This looks good on first glance but I'm gonna come back to it a second time when I'm not so busy.
> 
> If @0xBigBoss has time for review then we can merge
> 
> This is a staple function in the codebase and I need more time to read it and learn it

use compare with current implementation, small changes
image

**Comment by youngkidwarrior** [comment]
We aren't checking if the X username is unique and accurate. Why add this to profile_lookup?

**Comment by youngkidwarrior** [comment]
Please make sure to reorder this migration before merging it!

**Review by youngkidwarrior — APPROVED** [review]

**Pull request merged**